pennylane/templates/embeddings/iqp.py

```diff
@@ -257,3 +275,57 @@ def compute_decomposition(
                 op_list.append(MultiRZ(features[idx1] * features[idx2], wires=wire_pair))
 
         return op_list
+
+
+def _iqp_embedding_resources(pattern_size, num_repeats, num_wires):
+    return {
+        resource_rep(RZ): num_repeats * num_wires,
+        resource_rep(H): num_repeats * num_wires,
+        resource_rep(MultiRZ, num_wires=2): pattern_size * num_repeats,
+    }
+
+
+@register_resources(_iqp_embedding_resources)
+def _iqp_embedding_decomposition(features, wires, n_repeats, pattern):
+
+    if has_jax and capture.enabled():
+        wires, pattern, features = jnp.array(wires), jnp.array(pattern), jnp.array(features)
+
+    if math.ndim(features) > 1:
+        features = math.T(features)
+
+    @for_loop(n_repeats)
+    def outer_loop(_):
+
+        @for_loop(len(wires))
+        def inner_loop(i):
+            H(wires=wires[i])
+            RZ(features[i], wires=wires[i])
+
+        inner_loop()  # pylint: disable=no-value-for-parameter
```

Comment:
I would find this even more descriptive, super optional though.
```suggestion
        def single_qubit_loop(i):
            H(wires=wires[i])
            RZ(features[i], wires=wires[i])

        single_qubit_loop()  # pylint: disable=no-value-for-parameter
```